src/main/java/sg/ncl/MainController.java

```diff
@@ -4041,32 +4023,115 @@ public String emailChecklist(@ModelAttribute("statuschecklist") String status) {
+                            log.error(error.getMessage());
+                            redirectAttributes.addFlashAttribute(MESSAGE, "Adding of public key is forbidden");
+                            break;
+                        default:
+                            log.error("Unknown error when adding public key");
+                            redirectAttributes.addFlashAttribute(MESSAGE, "Unknown error when adding public key");
+                    }
+                }
+            } catch (IOException e) {
+                throw new WebServiceRuntimeException(e.getMessage());
+            }
+        }
+
+        return "redirect:/show_pub_keys";
+    }
+
+    @RequestMapping(path = "/delete_pub_key/{keyId}", method = RequestMethod.GET)
+    public String deletePublicKey(HttpSession session, @PathVariable String keyId) throws WebServiceRuntimeException {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(
+                properties.getPublicKeys(session.getAttribute("id").toString()) + "/" + keyId,
+                HttpMethod.DELETE, request, String.class);
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                log.error("Unable to delete public key {} for user {}", keyId, session.getAttribute("id"));
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                throw new RestClientException("[" + error.getError() + "] ");
+            }
+        } catch (IOException e) {
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return "redirect:/show_pub_keys";
+    }
+
     //--------------------------Get List of scenarios filenames--------------------------
-    private List<String> getScenarioFileNameList() throws WebServiceRuntimeException {
+    private List<String> getScenarioFileNameList() {
         log.info("Retrieving scenario file names");
-//        List<String> scenarioFileNameList = null;
-//        try {
-//            scenarioFileNameList = IOUtils.readLines(getClass().getClassLoader().getResourceAsStream("scenarios"), StandardCharsets.UTF_8);
-//        } catch (IOException e) {
-//            throw new WebServiceRuntimeException(e.getMessage());
-//        }
-//        File folder = null;
-//        try {
-//            folder = new ClassPathResource("scenarios").getFile();
-//        } catch (IOException e) {
-//            throw new WebServiceRuntimeException(e.getMessage());
-//        }
-//        List<String> scenarioFileNameList = new ArrayList<>();
-//		File[] files = folder.listFiles();
-//		for (File file : files) {
-//			if (file.isFile()) {
-//				scenarioFileNameList.add(file.getError());
-//			}
-//		}
         // FIXME: hardcode list of filenames for now
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)